packages/@aws-cdk/cloudformation-include/test/valid-templates.test.ts

```diff
@@ -236,63 +255,77 @@ describe('CDK Include', () => {
   test('can ingest a template with intrinsic functions and conditions, and output it unchanged', () => {
     includeTestTemplate(stack, 'functions-and-conditions.json');
 
-    expect(stack).toMatchTemplate(
+    Template.fromStack(stack).templateMatches(
       loadTestFileToJsObject('functions-and-conditions.json'),
     );
   });
 
   test('can ingest a JSON template with string-form Fn::GetAtt, and output it unchanged', () => {
     includeTestTemplate(stack, 'get-att-string-form.json');
 
-    expect(stack).toMatchTemplate(
+    Template.fromStack(stack).templateMatches(
       loadTestFileToJsObject('get-att-string-form.json'),
     );
   });
 
   test('can ingest a template with Fn::Sub in string form with escaped and unescaped references and output it unchanged', () => {
     includeTestTemplate(stack, 'fn-sub-string.json');
 
-    expect(stack).toMatchTemplate(
+    Template.fromStack(stack).templateMatches(
       loadTestFileToJsObject('fn-sub-string.json'),
     );
   });
 
   test('can parse the string argument Fn::Sub with escaped references that contain whitespace', () => {
     includeTestTemplate(stack, 'fn-sub-escaping.json');
 
-    expect(stack).toMatchTemplate(
+    Template.fromStack(stack).templateMatches(
       loadTestFileToJsObject('fn-sub-escaping.json'),
     );
   });
 
   test('can ingest a template with Fn::Sub in map form and output it unchanged', () => {
     includeTestTemplate(stack, 'fn-sub-map-dotted-attributes.json');
 
-    expect(stack).toMatchTemplate(
-      loadTestFileToJsObject('fn-sub-map-dotted-attributes.json'),
-    );
+    Template.fromStack(stack).hasResourceProperties('AWS::S3::Bucket', {
+      "BucketName": {
+        "Fn::Sub": "${ELB.SourceSecurityGroup.GroupName}",
+      },
+    });
+
+    Template.fromStack(stack).hasResourceProperties('AWS::ElasticLoadBalancing::LoadBalancer', {
+      "AvailabilityZones": [
+        "us-east-1a",
+      ],
+      "CrossZone": true,
+      "Listeners": [{
+        "LoadBalancerPort": "80",
+        "InstancePort": "80",
+        "Protocol": "HTTP",
+      }],
+    });
```

Comment:
So, the name of the test is "can ingest a template with Fn::Sub in map form and output it unchanged".

And yet, none of the assertions verify there is an `Fn::Sub` with a map.

Seems off a little bit...?